contracts/libraries/HyperdriveExecution.sol

```diff
@@ -0,0 +1,578 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+import { IHyperdrive } from "hyperdrive/contracts/src/interfaces/IHyperdrive.sol";
+import { FixedPointMath } from "hyperdrive/contracts/src/libraries/FixedPointMath.sol";
+import { HyperdriveMath } from "hyperdrive/contracts/src/libraries/HyperdriveMath.sol";
+import { SafeCast } from "hyperdrive/contracts/src/libraries/SafeCast.sol";
+import { YieldSpaceMath } from "hyperdrive/contracts/src/libraries/YieldSpaceMath.sol";
+import { Packing } from "openzeppelin/utils/Packing.sol";
+import { IEverlong } from "../interfaces/IEverlong.sol";
+import { IEverlongEvents } from "../interfaces/IEverlongEvents.sol";
+
+// TODO: Extract into its own library.
+uint256 constant HYPERDRIVE_SHARE_RESERVES_BOND_RESERVES_SLOT = 2;
+uint256 constant HYPERDRIVE_LONG_EXPOSURE_LONGS_OUTSTANDING_SLOT = 3;
+uint256 constant HYPERDRIVE_SHARE_ADJUSTMENT_SHORTS_OUTSTANDING_SLOT = 4;
+
+/// @author DELV
+/// @title HyperdriveExecutionLibrary
+/// @notice Library to handle the execution of trades with hyperdrive.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+library HyperdriveExecutionLibrary {
+    using FixedPointMath for uint256;
+    using SafeCast for *;
+    using Packing for bytes32;
+
+    // ╭─────────────────────────────────────────────────────────╮
+    // │ Open Long                                               │
+    // ╰─────────────────────────────────────────────────────────╯
+
+    /// @notice Opens a long with hyperdrive using amount.
+    /// @param _asBase Whether to use hyperdrive's base asset.
+    /// @param _amount Amount of assets to spend.
+    /// @return maturityTime Maturity timestamp of the opened position.
+    /// @return bondAmount Amount of bonds received.
+    function openLong(
+        IHyperdrive self,
+        bool _asBase,
+        uint256 _amount
+    ) internal returns (uint256 maturityTime, uint256 bondAmount) {
+        (maturityTime, bondAmount) = self.openLong(
+            _amount,
+            0,
+            0,
```

Comment:
We might add a `TODO` here. I could see it being useful to have slippage/negative interest protection, but I could be missing why we don't want to do this here.

Reply:
With regards to slippage, I was under the impression from our conversation earlier in the week that there is no slippage in an atomic scenario, and that the parameter is intended for MEV protection. Can you help clarify this?

Will add a todo wrt vaultshareprice.

Reply:
Good question, here's how I'd frame this. There is no slippage from the time of valuing positions to the time of opening the trade. That said, from the perspective of someone that wants to withdraw from Everlong, there may be slippage between sending out the transaction and the transaction actually being executed. I'm envisioning that this would be a protection for the user that is withdrawing. We could also implement this higher up in the stack though with some kind of `minimumPricePerShare`, which may make more sense in this context.

Reply:
added a todo